replica/table.cc

```diff
@@ -1478,11 +1478,16 @@ future<std::unordered_map<sstring, table::snapshot_details>> table::get_snapshot
                 continue;
             }
 
-            lister::scan_dir(snapshots_dir,  { directory_entry_type::directory }, [this, datadir, &all_snapshots] (fs::path snapshots_dir, directory_entry de) {
+            lister::scan_dir(snapshots_dir,  { directory_entry_type::directory }, [this, datadir, &all_snapshots] (fs::path snapshots_dir, directory_entry de) -> future<> {
                 auto snapshot_name = de.name;
-                all_snapshots.emplace(snapshot_name, snapshot_details());
-                return lister::scan_dir(snapshots_dir / fs::path(snapshot_name),  { directory_entry_type::regular }, [this, datadir, &all_snapshots, snapshot_name] (fs::path snapshot_dir, directory_entry de) {
-                    return io_check(file_size, (snapshot_dir / de.name).native()).then([this, datadir, &all_snapshots, snapshot_name, snapshot_dir, name = de.name] (auto size) {
+                auto snapshot_dir = snapshots_dir / snapshot_name;
+                auto sd = co_await io_check(file_stat, snapshot_dir.native(), follow_symlink::no);
+                all_snapshots.emplace(snapshot_name, snapshot_details{sd.allocated_size, sd.allocated_size});
```

Comment:
I think that this might cause the 4096 value returning for an otherwise empty snapshot directory (on filesystems that might count the dir inode itself?).
Previously, the snapshot details didn't account for the directory itself so we can keep it this way.

Reply:
I removed it and testing a new version.
Will push as soon as it passes unit tests

Reply:
Good. As you noticed, in Posix the "size" of a directory is basically meaningless. On btrfs an empty directory has size 0, but on other filesystems it can be anything - 40 or 4096 or 1234. In classic Unix filesystems, directories were stored as a normal file on disk containing a list - and later some sort of tree implementation - of files, so even an empty directory wasn't empty because it always contained entries for "." and "..".